java-checks/src/main/java/org/sonar/java/checks/AssertionsInTestsCheck.java

```diff
@@ -123,109 +108,35 @@ public void visitMethod(MethodTree methodTree) {
     boolean isUnitTest = isUnitTest(methodTree);
     inUnitTest.push(isUnitTest);
     methodContainsAssertion.push(false);
-    methodContainsAssertjSoftAssertionUsage.push(false);
-    methodContainsAssertjAssertAll.push(false);
-    methodContainsJunitSoftAssertionUsage.push(false);
     super.visitMethod(methodTree);
-    inUnitTest.pop();
     Boolean containsAssertion = methodContainsAssertion.pop();
-    Boolean containsSoftAssertionDecl = methodContainsAssertjSoftAssertionUsage.pop();
-    Boolean containsAssertjAssertAll = methodContainsAssertjAssertAll.pop();
-    Boolean containsJunitSoftAssertionUsage = methodContainsJunitSoftAssertionUsage.pop();
-    if (isUnitTest &&
-        !expectAssertion(methodTree) &&
-        (!containsAssertion || badSoftAssertionUsage(containsSoftAssertionDecl, containsAssertjAssertAll, containsJunitSoftAssertionUsage))) {
+    inUnitTest.pop();
+    if (isUnitTest && !expectAssertion(methodTree) && !containsAssertion) {
       context.reportIssue(this, methodTree.simpleName(), "Add at least one assertion to this test case.");
     }
   }
 
-  private static boolean expectAssertion(MethodTree methodTree) {
-    List<SymbolMetadata.AnnotationValue> annotationValues = methodTree.symbol().metadata().valuesForAnnotation("org.junit.Test");
-    if (annotationValues != null) {
-      for (SymbolMetadata.AnnotationValue annotationValue : annotationValues) {
-        if ("expected".equals(annotationValue.name())) {
-          return true;
-        }
-      }
-    }
-    return false;
-  }
-
-  private static boolean badSoftAssertionUsage(Boolean containsSoftAssertionDecl, Boolean containsAssertjAssertAll, Boolean containsJunitSoftAssertionUsage) {
-    return containsSoftAssertionDecl && !containsJunitSoftAssertionUsage && !containsAssertjAssertAll;
-  }
-
   @Override
   public void visitMethodInvocation(MethodInvocationTree mit) {
-    if (!inUnitTest()) {
-      return;
-    }
-    checkForAssertjSoftAssertions(mit);
-    if (methodContainsAssertion.peek()) {
-      return;
-    }
-    chainedTo.push(ChainedMethods.NONE);
     super.visitMethodInvocation(mit);
-    ChainedMethods chainedToResult = chainedTo.pop();
-    if (containsAssertion(mit, chainedToResult)) {
-      methodContainsAssertion.pop();
-      methodContainsAssertion.push(Boolean.TRUE);
-    }
-    if (!chainedTo.isEmpty()) {
-      if (ChainedMethods.ASSERT_THAT.equals(chainedToResult) || ASSERT_THAT.matches(mit)) {
-        chainedTo.pop();
-        chainedTo.push(ChainedMethods.ASSERT_THAT);
-      } else if (MOCKITO_VERIFY.matches(mit)) {
-        chainedTo.pop();
-        chainedTo.push(ChainedMethods.MOCKITO_VERIFY);
+    if (!methodContainsAssertion.peek() && inUnitTest.peek()) {
+      if (ASSERTION_INVOCATION_MATCHERS.anyMatch(mit)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1066)